book/03-git-branching/sections/branch-management.asc

```diff
@@ -73,3 +73,42 @@ $ git branch --no-merged master
   featureB
 ----
 ====
+
+==== Changing a branch name
+
+You can change a branch name with the `git branch -m` command.
+If you want to collaborate with others, on the newly changed branch, you'll need to push it.
+
+[source,console]
+----
+$ git branch -m bad-branch-name corrected-branch-name
+$ git push -u origin corrected-branch-name
```

Comment:
I think the end result here would be that both `bad-branch-name` and `corrected-branch-name` exist on the remote? `git branch -m` only renames the branch locally, so we may want to get a bit more in depth there — the concept that my branches and the remote's branches can be different that trips up a lot of newcomers.

Reply:
Agree, I'll go expand on this some more, and try a move on a new test repo to find out how the remote and local branch interact. That way I can document what happens.

Reply:
>I think the end result here would be that both bad-branch-name and corrected-branch-name exist on the remote? 

Yes that is correct, I've just tested it.
The local bad branch is renamed, then you push the local correct branch up.
The remote now has both branches, both the correct one and the bad one.
The user then needs to `push origin --delete bad-branch-name` to get rid of the bad branch.


> so we may want to get a bit more in depth there — the concept that my branches and the remote's branches can be different that trips up a lot of newcomers.

I've added a overview of the branches just after pushing the correct branch, I hope this helps explain what just happened.

I'm marking this as resolved, because the review is now out of date. We'll continue this conversation on the newer version of the text 😉 .